provider/cluster/lease_withdraw.go

```diff
@@ -0,0 +1,96 @@
+package cluster
+
+import (
+	"context"
+	lifecycle "github.com/boz/go-lifecycle"
+	"github.com/ovrclk/akash/provider/event"
+	"github.com/ovrclk/akash/provider/session"
+	"github.com/ovrclk/akash/pubsub"
+	"github.com/ovrclk/akash/util/runner"
+	mtypes "github.com/ovrclk/akash/x/market/types"
+	"github.com/tendermint/tendermint/libs/log"
+	"time"
+)
+
+type deploymentWithdrawal struct {
+	bus     pubsub.Bus
+	session session.Session
+	lease   mtypes.LeaseID
+	log     log.Logger
+	lc      lifecycle.Lifecycle
+}
+
+func newDeploymentWithdrawal(dm *deploymentManager) *deploymentWithdrawal {
+	m := &deploymentWithdrawal{
+		bus:     dm.bus,
+		session: dm.session,
+		lease:   dm.lease,
+		log:     dm.log.With("cmp", "deployment-withdrawal"),
+		lc:      lifecycle.New(),
+	}
+
+	go m.lc.WatchChannel(dm.lc.ShuttingDown())
+	go m.run()
+
+	return m
+}
+
+func (dw *deploymentWithdrawal) doWithdrawal(ctx context.Context) error {
+	msg := &mtypes.MsgWithdrawLease{
+		LeaseID: dw.lease,
+	}
+
+	return dw.session.Client().Tx().Broadcast(ctx, msg)
+}
+
+func (dw *deploymentWithdrawal) run() {
+	defer dw.lc.ShutdownCompleted()
+	ctx, cancel := context.WithCancel(context.Background())
+
+	events, err := dw.bus.Subscribe()
+	if err != nil {
+		dw.log.Error("Could not subscribe to events", "err", err)
+	}
+
+	const withdrawalPeriod = 24 * time.Hour
```

Comment:
I really miss having https://github.com/caarlos0/env for settings so that we could change these kinds of things without creating a new release.